File: java/src/main/java/deltix/luminary/Float32Def.java
package deltix.luminary;

import java.util.ArrayList;
import java.util.List;

public class Float32Def implements TypeDef<TypeFloat32>, ConstantScope {
    private List<ConstantDef> definedConstants;

    private Float32Def() {
        definedConstants = new ArrayList<>();
        definedConstants.add(new ConstantDef(this, "MIN_VALUE", TypeFloat32.INSTANCE, new LiteralFloat32(Float.MIN_VALUE)));
        definedConstants.add(new ConstantDef(this, "MAX_VALUE", TypeFloat32.INSTANCE, new LiteralFloat32(Float.MAX_VALUE)));
        definedConstants.add(new ConstantDef(this, "NaN", TypeFloat32.INSTANCE, new LiteralFloat32(Float.NaN)));
        definedConstants.add(new ConstantDef(this, "POSITIVE_INFINITY", TypeFloat32.INSTANCE, new LiteralFloat32(Float.POSITIVE_INFINITY)));
        definedConstants.add(new ConstantDef(this, "NEGATIVE_INFINITY", TypeFloat32.INSTANCE, new LiteralFloat32(Float.NEGATIVE_INFINITY)));
    }

    public static final Float32Def INSTANCE = new Float32Def();

    @Override
    public String getName() {
        return "Float32";
    }

    @Override
    public String getFullName() {
        return getName();
    }

    @Override
    public List<ConstantDef> getDefinedConstants() {
        return definedConstants;
    }

    @Override
    public TypeFloat32 getType() {
        return TypeFloat32.INSTANCE;
    }

    @Override
    public FileDef getFile() {
        return null;
    }

    @Override
    public TypeScope getParent() {
        return null;
    }
}

File: csharp/main/Float32Def.cs
using System;
using System.Collections.Generic;

namespace Deltix.Luminary
{
	public class Float32Def : IConstantScope, ITypeDef<TypeFloat32>
	{
		private Float32Def()
		{
			DefinedConstants.Add(new ConstantDef(this, "MIN_VALUE", TypeFloat32.Instance, new LiteralFloat32(Single.MinValue)));
			DefinedConstants.Add(new ConstantDef(this, "MAX_VALUE", TypeFloat32.Instance, new LiteralFloat32(Single.MaxValue)));
			DefinedConstants.Add(new ConstantDef(this, "NaN", TypeFloat32.Instance, new LiteralFloat32(Single.NaN)));
			DefinedConstants.Add(new ConstantDef(this, "POSITIVE_INFINITY", TypeFloat32.Instance, new LiteralFloat32(Single.PositiveInfinity)));
			DefinedConstants.Add(new ConstantDef(this, "NEGATIVE_INFINITY", TypeFloat32.Instance, new LiteralFloat32(Single.NegativeInfinity)));
		}

		public static Float32Def Instance = new Float32Def();

		public String Name => "Float32";

		public String FullName => Name;

		public List<ConstantDef> DefinedConstants { get; } = new List<ConstantDef>();

		public TypeFloat32 Type => TypeFloat32.Instance;

		public FileDef File => null;

		public ITypeScope Parent => null;
	}
}
